planner/core/logical_plan_builder.go

```diff
@@ -193,7 +193,9 @@ func (b *PlanBuilder) buildResultSetNode(ctx context.Context, node ast.ResultSet
 			if name.Hidden {
 				continue
 			}
-			name.OrigTblName = name.TblName
+			if name.OrigTblName.L == "" {
```

Comment:
Do we really need this if branch?
I think if the resultSet is a virtual table. We don't need set the original table name.